app/uk/gov/hmrc/cipphonenumberverification/services/VerifyHelper.scala

```diff
@@ -0,0 +1,94 @@
+    val sb = new StringBuilder()
+    val passcodeSize = 6
+    val chrsToChooseFrom = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    val secureRandom = SecureRandom.getInstanceStrong
+    secureRandom.ints(passcodeSize, 0, chrsToChooseFrom.length)
+      .mapToObj((i: Int) => chrsToChooseFrom.charAt(i))
+      .forEach(x => sb.append(x))
+    sb.mkString
+  }
+
+  private[services] def persistPasscode(phoneNumber: PhoneNumber)(implicit hc: HeaderCarrier) = {
+    logger.debug(s"Storing passcode in database for ${phoneNumber.phoneNumber}")
+    val otp = otpGenerator
+    val passcode = Passcode(phoneNumber.phoneNumber, otp)
+    passcodeCacheRepository.persistPasscode(phoneNumber, passcode)
+  }
+
+  private[services] def retrievePasscode(passcode: Passcode) = {
+    logger.debug(s"Retrieving passcode from database for ${passcode.phoneNumber}")
+    passcodeCacheRepository.get[Passcode](passcode.phoneNumber)(DataKey("cip-phone-number-verification"))
+  }
+
+  private[services] def delete(passcode: Passcode) = {
+    logger.debug(s"Deleting passcode from database for ${passcode.phoneNumber}")
+    passcodeCacheRepository.delete(passcode.phoneNumber)(DataKey("cip-phone-number-verification"))
+  }
+
+  private[services] def processResponse(res: HttpResponse,  phoneNumber: PhoneNumber)(implicit hc: HeaderCarrier) = res match {
+    case _ if is2xx(res.status) => processPhoneNumber(res, phoneNumber)
+    case _ if is4xx(res.status) => Future(BadRequest(res.json))
+  }
+
+  private[services] def processPhoneNumber(res: HttpResponse, phoneNumber: PhoneNumber)(implicit hc: HeaderCarrier) = (res.json \ "phoneNumberType").as[String] match {
+    case "Mobile" =>
+      persistPasscode(phoneNumber) flatMap sendPasscode recover {
+        case err =>
+          logger.error(s"Database operation failed - ${err.getMessage}")
+          InternalServerError(Json.toJson(ErrorResponse("DATABASE_OPERATION_FAIL", "Database operation failed")))
+      }
+    case _ => Future(Ok(Json.toJson(IndeterminateResponse("Indeterminate", "Only mobile numbers can be verified"))))
+  }
+
+  private[services] def sendPasscode(passcode: Passcode)(implicit hc: HeaderCarrier) = (govUkConnector.sendPasscode(passcode) map {
+    case Left(err) => ??? //TODO: CAV-163
+      logger.error(s"Gov Notify failure - to be covered by CAV-163")
+      InternalServerError(Json.toJson(ErrorResponse("EXTERNAL_SYSTEM_FAIL", "sending to Gov Notify failed")))
+    case Right(response) if response.status == 201 => Accepted(Json.parse(s"""{"notificationId" : ${response.json("id")}}"""))
+  }) recover {
+    case err =>
+      logger.error(s"GOVNOTIFY operation failed - ${err.getMessage}")
+      InternalServerError(Json.toJson(ErrorResponse("GOVNOTIFY_OPERATION_FAIL", "Gov Notify operation failed")))
+  }
+
+  private[services] def processPasscode(passcode: Passcode, pass: Option[Passcode]) = pass match {
+    case Some(actualPasscode) if passcode.otp.equals(actualPasscode.otp) => delete(passcode) map {
```

Comment:
put the does it match logic in a separate class.  as this may change in the future.  also its business logic and not a db operation